pkg/cloud/services/compute/instance.go

```diff
@@ -504,6 +505,25 @@ func (s *Service) getOrCreatePort(eventObject runtime.Object, clusterName string
 		return nil, err
 	}
 
+	if portOpts.PortSecurity != nil {
+		portUpdateOpts := ports.UpdateOpts{}
+		if *portOpts.PortSecurity == false {
+			// Remove allowed address pairs and security groups from port if possible to avoid openstack errors
+			// when port security is disabled on a port.
+			portUpdateOpts.SecurityGroups = &[]string{}
+			portUpdateOpts.AllowedAddressPairs = &[]ports.AddressPair{}
+		}
+		updateOpts := portsecurity.PortUpdateOptsExt{
+			UpdateOptsBuilder:   portUpdateOpts,
+			PortSecurityEnabled: portOpts.PortSecurity,
+		}
+		port, err = ports.Update(s.networkClient, port.ID, updateOpts).Extract()
```

Comment:
if we really need use `update` , we need add metrics like line 503 above

Reply:
I don't think we need this Update. I can't see why we wouldn't just pass PortSecurityEnabled during the creation above. We're hitting neutron twice for no obvious reason.

Reply:
The only reason would be to set an existing OpenStackMachine to enable or disable port security. But I don't know if this is even allowed (immutability?).

Reply:
Is it possible to do a **create** call in gophercloud with more than one extension applied?

Reply:
opened a question https://github.com/gophercloud/gophercloud/issues/2195 ,hopefully we can get some comments